### PR TITLE
feat: add config refetching when SSE message received

### PR DESCRIPTION
**Pull request by ajwootto**
- on receiving message from streaming connection, fetch new config
- refactor stuff in Client class to reduce duplication of handling a new config being received
- rewrite RequestConsolidator to avoid sending redundant config requests when duplicate calls are made, and prevent config re-fetch due to SSE from overwriting the latest identified user if there's an ongoing request.

### CODE REVIEW

**Comment by ajwootto** [comment]
Current dependencies on/for this PR:
* main
  * **PR #280** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/280" target="_blank">Graphite</a> 
    * **PR #281** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/281" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/281?utm_source=stack-comment).

File: sdk/js/src/ConfigRequestConsolidator.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        return resolver
+    }
+
+    private async processQueue() {
+        if (!this.resolvers.length) {
+            return
+        }
+
+        const resolvers = this.resolvers.splice(0)
+        const bucketedConfig = await this.performRequest(this.nextUser).then((result) => {
```
Comment (elliotCamblor) [inline sdk/js/src/ConfigRequestConsolidator.ts:47]: do you need to return `result` somewhere in this callback?

Comment (ajwootto) [reply sdk/js/src/ConfigRequestConsolidator.ts:47]: yeah in order for the return value to be correct, you're right.
 However I'm not actually using the return value of this method so I might just have it return nothing

Comment (elliotCamblor) [reply sdk/js/src/ConfigRequestConsolidator.ts:47]: by callback I mean the `.then`, im just not entirely sure how the bucketedConfig variable is going to be set, its kinda making my head spin

Comment (elliotCamblor) [reply sdk/js/src/ConfigRequestConsolidator.ts:47]: oh I see what youre saying, that makes sense

**Review by elliotCamblor — APPROVED** [review]

**Review by nsalamad — APPROVED** [review]

**Pull request merged**